File: src/PiecykPolHurt/PiecykPolHurt.Model/Dto/Report/ReportParamDto.cs
using PiecykPolHurt.Model.Enums;

namespace PiecykPolHurt.Model.Dto.Report
{
    public class ReportParamDto
    {
        public string Name { get; set; }
        public string Caption { get; set; }
        public ParamType Type { get; set; }
        public bool IsRequired { get; set; }
        public IEnumerable<ReportParamValueDto> AvailableValues { get; set; }
    }
}

File: src/piecyk-pol-hurt-frontend/src/API/Models/Reports/ReportParamDto.ts
import { ParamType } from "./ParamType";

export interface ReportParamDto {
  name: string;
  caption: string;
  type: ParamType;
  isRequired: boolean;
  availableValues: ReportParamValueDto[];
}

interface ReportParamValueDto {
  label: string;
  value: string;
}
